sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -174,11 +180,28 @@ namespace Azure { namespace Core { namespace Http {
       }
     };
 
-    /**
-     * @brief libcurl handle to be used in the session.
-     *
-     */
-    CURL* m_pCurl;
+    // connection handle. It will be taken from a pool
+    class CurlConnection {
+    private:
+      CURL* m_handle;
+      std::string m_host;
+
+    public:
+      CurlConnection(std::string const& host) : m_handle(curl_easy_init()), m_host(host) {}
+
+      ~CurlConnection() { curl_easy_cleanup(this->m_handle); }
+
+      CURL* GetHandle() { return this->m_handle; }
+
+      std::string GetHost() const { return this->m_host; }
+    };
+
+    // TODO: Mutex for this code to access connectionPool
+    static std::list<std::unique_ptr<CurlConnection>> c_connectionPool;
```

Comment:
@vhvb1989 Please do confirm with Casey that static variables won't cause any issue for dll.